rbac/management/role/view.py

```diff
@@ -336,27 +362,13 @@ def destroy(self, request, *args, **kwargs):
             HTTP/1.1 204 NO CONTENT
         """
         validate_uuid(kwargs.get("uuid"), "role uuid validation")
-        role = self.get_object()
-        if role.system or role.platform_default:
-            key = "role"
-            message = "System roles cannot be deleted."
-            error = {key: [_(message)]}
-            raise serializers.ValidationError(error)
+
         try:
             with transaction.atomic():
-                self.delete_policies_if_no_role_attached(role)
-                dual_write_handler = RelationApiDualWriteHandler(role, "DELETE")
-                dual_write_handler.generate_relations_from_current_state_of_role()
                 response = super().destroy(request=request, args=args, kwargs=kwargs)
-                dual_write_handler.save_replication_event_to_outbox()
         except DualWriteException as e:
             return self.dual_write_exception_response(e)
 
-        if response.status_code == status.HTTP_204_NO_CONTENT:
-            role_obj_change_notification_handler(role, "deleted", request.user)
-
-            auditlog = AuditLog()
-            auditlog.log_delete(request, AuditLog.ROLE, role)
```

Comment:
Audit log stuff seemed to make more sense in the `perform_` hooks now that we added those, since they already have access to the latest instance. But happy to revert back if desired.